### PR TITLE
feat(dropdown): update style vue3

### PR DESCRIPTION
**Pull request by francisrupert**
# PR Title

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

Style updates to Dropdown, and one minor update to List item

## :bulb: Context

Dialtone 7 style uptake

## :pencil: Checklist

- [x] I have reviewed my changes
- [ ] I have added tests
- [ ] I have added all relevant documentation
- [x] I have validated components with a screen reader
- [x] I have validated components keyboard navigation
- [x] I have considered the performance impact of my change
- [x] I have checked that my change did not significantly increase bundle size
- [ ] I am exporting any new components or constants in the index.js in the component directory
- [ ] I am exporting any new components or constants in the index.js in the root

## :camera: Screenshots / GIFs

image

### CODE REVIEW

File: components/dropdown/dropdown_variants.story.vue [hunk 1]
```diff
@@ -120,8 +120,8 @@
           #list="{ close }"
         >
           <dt-list-item-group
-            heading-class="d-p6 d-fw-bold"
-            :list-class="$attrs.listClass"
+            heading-class="d-py4 d-px8 d-fw-semibold d-c-default"
+            :list-class="listClass"
```
Comment (braddialpad) [inline components/dropdown/dropdown_variants.story.vue]: This one does in fact have to be `$attrs.listClass` in vue 3

File: components/dropdown/dropdown_variants.story.vue [hunk 2]
```diff
@@ -142,8 +142,8 @@
           </dt-list-item-group>
           <dt-dropdown-separator />
           <dt-list-item-group
-            :list-class="$attrs.listClass"
-            heading-class="d-p6 d-fw-bold"
+            :list-class="listClass"
```
Comment (braddialpad) [inline components/dropdown/dropdown_variants.story.vue]: `$attrs.listClass`

**Review by braddialpad — CHANGES_REQUESTED** [review]

File: components/dropdown/dropdown_variants.story.vue (hunk 2 above)
Comment (francisrupert) [reply components/dropdown/dropdown_variants.story.vue]: done

**Review by braddialpad — APPROVED** [review]
Looks good!

**Pull request merged**